cpp/include/cudf/detail/structs/utilities.hpp

```diff
@@ -167,7 +167,7 @@ class flattened_table {
  * @return `flatten_result` with flattened table, flattened column order, flattened null precedence,
```

Comment:
the docs mention `flattened_result` in a few places, but I don't see this type in the code. Would be nice to clean this up in this PR, since it touches the same files/functions.